conan/tools/cmake/toolchain.py

```diff
@@ -356,7 +356,10 @@ def _apple_sdk_name(self):
         we're building for (considering simulators)"""
         arch = self._conanfile.settings.get_safe('arch')
         os_ = self._conanfile.settings.get_safe('os')
-        if arch.startswith('x86'):
+        os_sdk = self._conanfile.settings.get_safe('os.sdk')
+        if os_sdk:
+            return os_sdk
+        elif arch.startswith('x86'):
```

Comment:
Better force the definition of ``os.sdk``?

The thing is that we don't want to auto-deduce it is a simulator. The binary is different, it should have a different package_id, and for that it should have the setting properly defined. It might not be robust relying on the architecture to deduce this. 

It might be necessary to implement the logic of ``Macos`` => ``macosx`` as there is no other sdk for it, but for iOS, watchOS, tvOS, it should be mandatory to be defined in settings.

Reply:
Yeah, you're right but it sounds like a breaking change, doesn't it? Should we make that change in another PR?

Reply:
Or could we even introduce that change for Conan 2.x?

Reply:
This is ``CMakeToolchain``, which is experimental. It might be fine to make it only work with the 2.0-ready behavior, and fail otherwise, even if it is "breaking" (I don't think it is a huge issue, as it boils down to add the sdk in your profiles or command line, not needing to modify recipes, and it is not that CMakeToolchain for iOS is that extended yet either.

Reply:
Definitely, yes, let's do that for all the cases 1.x and 2.x.